### PR TITLE
fix(si): Change the si launch command to use xdg open functionality

### PR DESCRIPTION
**Pull request by stack72**

### CODE REVIEW

File: lib/si-cli/src/cmd/launch.rs [hunk 1]
```diff
@@ -28,13 +29,22 @@ async fn invoke(launch_metrics: bool, web_host: String, web_port: u32) -> CliRes
...
-    println!("Opening URL: {}", path);
-    match open::that(path.clone()) {
-        Ok(()) => Ok(()),
-        Err(_err) => Err(SiCliError::FailToLaunch(path)),
+    let output = if cfg!(target_os = "macos") {
+        Command::new("open").arg(path.clone()).output()
+    } else if cfg!(target_os = "linux") {
+        Command::new("xdg-open").arg(path.clone()).output()
+    } else {
+        // This should NEVER get called but I added it in here just incase
```
Comment (nickgerace) [inline lib/si-cli/src/cmd/launch.rs:37]: Agreed!

File: lib/si-cli/src/cmd/launch.rs [hunk 2]
```diff
@@ -28,13 +29,22 @@ async fn invoke(launch_metrics: bool, web_host: String, web_port: u32) -> CliRes
...
+        return Err(SiCliError::UnsupportedOperatingSystem(
+            env::consts::OS.to_string(),
+        ));
+    };
+
+    if let Err(_err) = output {
```
Comment (nickgerace) [inline lib/si-cli/src/cmd/launch.rs:43]: I don't know if we have logging hooked up, but you could trace log it if so

**Review by paulocsanz — APPROVED** [review]

File: lib/si-cli/src/cmd/launch.rs [hunk 3]
```diff
@@ -28,13 +29,22 @@ async fn invoke(launch_metrics: bool, web_host: String, web_port: u32) -> CliRes
...
     }
 
-    println!("Opening URL: {}", path);
-    match open::that(path.clone()) {
-        Ok(()) => Ok(()),
-        Err(_err) => Err(SiCliError::FailToLaunch(path)),
+    let output = if cfg!(target_os = "macos") {
+        Command::new("open").arg(path.clone()).output()
+    } else if cfg!(target_os = "linux") {
+        Command::new("xdg-open").arg(path.clone()).output()
```
Comment (nickgerace) [inline lib/si-cli/src/cmd/launch.rs:35]: We could conditionally compile too.
Example:
```rust
#[cfg(any(
    all(target_os = "linux", target_arch = "x86_64"),
    all(target_os = "linux", target_arch = "aarch64")
))]
// do linux stuff

#[cfg(any(
    all(target_os = "macos", target_arch = "x86_64"),
    all(target_os = "macos", target_arch = "aarch64")
))]
// do macos stuff
```

**Comment by stack72** [comment]
bors merge

**Pull request merged**